extensions/wikia/MercuryApi/models/MercuryApi.class.php

```diff
@@ -248,9 +247,9 @@ private function clearUsers() {
 	 * @return string homepage URL. Default is US homepage.
 	 */
 	private function getHomepageUrl() {
-		global $wgLang;
+		global $wgContLang;
 		if ( class_exists('WikiaLogoHelper') ) {
-			return ( new WikiaLogoHelper() )->getCentralUrlForLang( $wgLang->getCode() );
+			return ( new WikiaLogoHelper() )->getCentralUrlForLang( $wgContLang->getCode() );
```

Comment:
@bognix We spoke with @rogatty and there's a ticket in the X-Wing backlog for making things straight with the user language. This is a fix for malfunctioning behavior cause by caching the WikiVariables response in iCache.